lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1511,6 +1522,116 @@ ClangImporter::emitBridgingPCH(StringRef headerPath,
   return false;
 }
 
+/// Determine the appropriate source input kind based on language options.
+static clang::Language getLanguageFromOptions(
+    const clang::LangOptions *LangOpts) {
+  if (LangOpts->OpenCL)
+    return clang::Language::OpenCL;
+  if (LangOpts->CUDA)
+    return clang::Language::CUDA;
+  if (LangOpts->ObjC)
+    return LangOpts->CPlusPlus ?
+        clang::Language::ObjCXX : clang::Language::ObjC;
+  return LangOpts->CPlusPlus ? clang::Language::CXX : clang::Language::C;
+}
+
+bool ClangImporter::emitPrecompiledModule(StringRef moduleMapPath,
```

Comment:
There is a significant amount of overlap between this and `ClangImporter::emitBridgingPCH`. Can we unify these two functions, so the somewhat arcane knowledge of how to get Clang to do this properly isn't duplicated?

Reply:
Done.